File: system-test/java/src/main/java/com/optivem/shop/testkit/port/given/steps/GivenPromotion.java
package com.optivem.shop.testkit.port.given.steps;

import com.optivem.shop.testkit.port.given.steps.base.GivenStep;

public interface GivenPromotion extends GivenStep {
    GivenPromotion withActive(boolean promotionActive);
    GivenPromotion withDiscount(double discount);
    GivenPromotion withDiscount(String discount);
}

File: system-test/typescript/src/testkit/dsl/port/given/steps/given-promotion.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenPromotion {
  withActive(promotionActive: boolean): this;
  withDiscount(discount: string | number): this;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}
